crates/ark/src/data_explorer/histogram.rs

```diff
@@ -0,0 +1,579 @@
+use std::collections::HashMap;
+
+use amalthea::comm::data_explorer_comm::ColumnFrequencyTable;
+use amalthea::comm::data_explorer_comm::ColumnFrequencyTableParams;
+use amalthea::comm::data_explorer_comm::ColumnHistogram;
+use amalthea::comm::data_explorer_comm::ColumnHistogramParams;
+use amalthea::comm::data_explorer_comm::ColumnHistogramParamsMethod;
+use amalthea::comm::data_explorer_comm::ColumnQuantileValue;
+use amalthea::comm::data_explorer_comm::FormatOptions;
+use anyhow::anyhow;
+use harp::exec::RFunction;
+use harp::exec::RFunctionExt;
+use harp::object::RObject;
+use harp::r_null;
+use harp::utils::r_classes;
+use harp::utils::r_inherits;
+use harp::utils::r_typeof;
+use libr::INTSXP;
+use libr::REALSXP;
+use libr::SEXP;
+use stdext::*;
+
+use crate::data_explorer::format::format_string;
+use crate::modules::ARK_ENVS;
+
+pub fn profile_histogram(
+    column: SEXP,
+    params: &ColumnHistogramParams,
+    format_options: &FormatOptions,
+) -> anyhow::Result<ColumnHistogram> {
+    let quantiles: RObject = match params.quantiles.clone() {
+        Some(v) => (&v).into(),
+        None => r_null().into(),
+    };
+
+    // Checks for supported objects:
+    // - Atomic integers and doubles
+    // - Dates and POSIXct objects
+    match r_classes(column) {
+        Some(v) => {
+            if !r_inherits(column, "Date") && !r_inherits(column, "POSIXct") {
+                return Err(anyhow!("Object with class '{:?}' unsupported.", v));
+            }
+        },
+        None => match r_typeof(column) {
+            INTSXP | REALSXP => {},
+            _ => return Err(anyhow!("Type not supported {:?}", r_typeof(column))),
+        },
+    }
+
+    let num_bins: RObject = match params.num_bins {
+        Some(v) => (v as i32).into(),
```

Comment:
Should we do these conversions with a tool that checks for boundaries? Not that it matters here but it seems like it'd be safer to get into a habit of checking bounds before truncating.